### DISCUSSION TITLE
How to make a POST request for Realtime Subscriptions Filter

### DISCUSSION THREAD

**Discussion by vsnshanmuganathan** [pocketbase/pocketbase#5629] [Q&A] [answered]
I successfully subscribed to a collection using a POST request with the following code: 
```json
{"clientId":"aAq9VBKbcWAk3MXXCxzio6xFaQDcTgdpT39ocrWw","subscriptions":["test"]}
```
and it worked perfectly.

I attempted to subscribe using filters, specifically for when the "name" column contains "line," but I still received all changes in the collection. I used the following code:

```json
{"clientId":"tKYQedPCNOrzxH1xFwfICDAAYGeJUcoTK2s7k1Rr","subscriptions":["test/?options={\"filter\": {\"name\": \"line\"}}"]}
```

However, it didn't work as intended.

Could you please assist me with how to make a POST request to subscribe using filters?

**Comment by ganigeorgiev** [comment, marked as answer]
`filter`, `expand`, `fields`, etc. are query parameters and they need to be set under the `query` topic options prop:

```js
?options={"query": {"filter": "name='test'"}}
```

Note that if you are using the JS SDK this is usually done through the optional SendOptions argument:

```js
await pb.collection("example").subscribe("*", (e) {
    // ...
}, {
    filter: "name='test'",
})
```

**Reply by vsnshanmuganathan** [reply]
I'm glad to hear that it works! Here’s the sample JSON I tested:

```json
{
  "clientId": "EitD63MfKV0m8AfRCaYhxtgbFtoDAvpJ5TSXwSHH",
  "subscriptions": [
    "test?options={\"query\": {\"filter\": \"name='test'\"}}"
  ]
}
```
